src/busctl/busctl.c

```diff
@@ -1691,9 +1692,25 @@ static int message_append_cmdline(sd_bus_message *m, const char *signature, char
                         break;
                 }
 
-                case SD_BUS_TYPE_UNIX_FD:
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "UNIX file descriptor not supported as type.");
+                case SD_BUS_TYPE_UNIX_FD: {
+                        int fd;
+
+                        fd = parse_fd(v);
+                        if (fd < 0)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse '%s' as a file descriptor", v);
```

Comment:
parse_fd() returns a negative error code in case of failure. This means we can pass `fd` as first argument to `log_error_errno()` here. i.e. if we receive an error from some function, then that's the error code we should pass on to the logging tooling

Reply:
oh, and if you receive an error code, always suffix the log message format string with `: %m` so that it is actually shown to the user, whatever it might be.